### PR TITLE
fix(date-picker): merge origin input and popup input focus

### PR DESCRIPTION
**Pull request by Yangzhedi**
## 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

fix[#11249](https://github.com/ant-design/ant-design/issues/11249)
add test for shared/focusTest.js

### CODE REVIEW

**Comment by Yangzhedi** [comment]
i add a `onFocus trigger only once` test
cased by 
```
Cascader › focus and blur › focus() and onFocus trigger only once

    expect(jest.fn()).toBeCalledTimes(1)

    Expected mock function to have been called one time, but it was called two times.

      35 |       wrapper.instance().focus();
      36 |       jest.runAllTimers();
    > 37 |       expect(handleFocus).toBeCalledTimes(1);
         |                           ^
      38 |     });
      39 | 
      40 |     it('blur() and onBlur', () => {

      at Object.toBeCalledTimes (tests/shared/focusTest.js:37:27)
```

File: components/date-picker/createPicker.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ export interface PickerProps {
 
 export interface PickerState {
   open: boolean;
+  hadFocus: boolean;
```
Comment (justjavac) [inline components/date-picker/createPicker.tsx]: Why not use `hasFocus`?

**Comment by afc163** [comment]
ci broken

**Comment by yoyo837** [comment]
ping

**Comment by afc163** [comment]
close for no response

**Pull request closed without merging**